contracts/javascore/xcall/src/main/java/foundation/icon/xcall/CallServiceImpl.java

```diff
@@ -155,20 +156,22 @@ public BigInteger sendCallMessage(String _to,
 
     @Override
     @External
-    public void executeCall(BigInteger _reqId) {
+    public void executeCall(BigInteger _reqId, byte[] _data) {
         CSMessageRequest req = proxyReqs.get(_reqId);
```

Comment:
how will dapp know about the original data? since we have replaced data with its hash? will they get it from event?

Reply:
Yes it should be in the CallMessage event